### DISCUSSION TITLE
How can I use the view(pairs) method in the Authoritative server to allocate based on the actual source IP of the request, instead of using the IP of the recursor server?

### DISCUSSION THREAD

**Discussion by jerryqzhang** [PowerDNS/pdns#13718] [Q&A] [answered]
- Program: Authoritative, Recursor 
 - Issue type: Feature request

### Short description

We want to dynamically resolve based on the netmask of the true source IP of the DNS request. Do you have any good suggestions?

### Usecase

We attempted to use the view method of the lua records function in Authoritative, but the remote IP obtained by Authoritative is the IP of the recursor server, not the actual source IP requested.

```
Authoritative: 4.8.3
```

https://doc.powerdns.com/authoritative/lua-records/functions.html

### Description

So how can we solve it? thank you very much

**Comment by mnordhoff** [comment, marked as answer]
If you enable the [`edns-subnet-processing` setting](https://doc.powerdns.com/authoritative/settings.html#edns-subnet-processing), the `view()` function will make use of the [EDNS Client Subnet extension](https://en.wikipedia.org/wiki/EDNS_Client_Subnet) **if** recursive resolvers send it (or use the resolver's IP if they do not).

If you're running a public service that random people across the Internet query, only some queries will include the extension: some resolvers always include it, some only include it if you make specific arrangements with the resolver operator, and many never include it at all.

If **you** run all of the resolvers at issue, you can enable it (such as with PowerDNS Recursor).

Edit: If you also use dnsdist, you need to configure it so that the ECS is cached and forwarded as you require.

**Reply by jerryqzhang** [reply]
Thank you very much for your help!
